### PR TITLE
Netatmo: Trigger reconnect cycle on auth errors from business API calls

### PR DESCRIPTION
**Pull request by Terdious**
### Pull Request check-list

To ensure your Pull Request can be accepted as fast as possible, make sure to review and check all of these items:

- [x] If your changes affects code, did your write the tests?
- [x] Are tests passing? (`npm test` on both front/server)
- [x] Is the linter passing? (`npm run eslint` on both front/server)
- [x] Did you run prettier? (`npm run prettier` on both front/server)
- [x] If you are adding a new features/services, did you run integration comparator? (`npm run compare-translations` on front)
- [x] Did you test this pull request in real life? With real devices? If this development is a big feature or a new service, we recommend that you provide a Docker image to the community ([forum](https://community.gladysassistant.com/)) for testing before merging.
- [ ] ~If your changes modify the API (REST or Node.js), did you modify the API documentation? (Documentation is based on comments in code)~
- [ ] ~If you are adding a new features/services which needs explanation, did you modify the user documentation? See [the GitHub repo](https://github.com/GladysAssistant/v4-website) and the [website](https://gladysassistant.com).~
- [ ] ~Did you add fake requests data for the demo mode (`front/src/config/demo.js`) so that the demo website is working without a backend? (if needed) See [https://demo.gladysassistant.com](https://demo.gladysassistant.com).~

NOTE: these things are not required to open a PR and can be done afterwards / while the PR is open.

## Summary

Follow-up to #2561. The OAuth refresh fix covers `/oauth2/token`, but business API calls (`homesdata`, `getthermostatsdata`, `getstationsdata`) also need to trigger the auto-reconnect cycle when Netatmo rejects the access token. Without this, clicking "Refresh" on the Discovery page when the Netatmo app is deactivated returned silently while the status stayed `connected` — confusing for the user.

> ⚠️ **Stacked on top of #2561.** Once #2561 is merged, this PR will rebase cleanly and the displayed diff will shrink to only the changes below.

## Details

- **New `handleApiAuthError(status)` helper.** On HTTP `401`, `403` or `406` (Netatmo's code for "Application is deactivated", confirmed by runtime testing: `406 { "error": { "code": 5, "message": "Application is deactivated" } }`), it switches the status to `RECONNECTING` and calls the `scheduleReconnectAttempt()` cycle introduced in #2561. Idempotent: no-op if a reconnect cycle is already in progress.
- **Wired into the 3 cloud loaders**: `loadDevices`, `loadThermostatDetails`, `loadWeatherStationDetails` — right after detecting a non-OK response.
- **`discoverDevices` and `refreshNetatmoValues`** no longer overwrite the status with `CONNECTED` when `handleApiAuthError` just moved it to `RECONNECTING`/`DISCONNECTED`. The UI keeps showing the auto-reconnect banner.
- **Tokens are never cleared by this helper itself.** `refreshingTokens` remains the single source of truth for token lifecycle decisions (24h grace window before clearing on fatal OAuth errors).

Runtime validated by the integration maintainer: deactivating the app on `dev.netatmo.com` while Gladys is up → click "Refresh" → status switches to `reconnecting`, banner appears in the 3 Netatmo pages, automatic retry cycles kick in; reactivating the app → reconnect succeeds at the next retry.

## Scope

| Area | Insertions | Deletions |
|---|---|---|
| Production code (server) | 67 | 3 |
| Tests | 159 | 0 |
| **Total vs #2561** | **226** | **3** |

Touched files reach 100% statements / branches / functions / lines coverage. Full netatmo suite: 197 passing.

## Test plan

- [x] `npm run test-service --service=netatmo` → 197 passing
- [x] 100% coverage on the 6 touched files
- [x] `npm run prettier` + `npm run eslint` on server (no new issues)
- [x] Runtime: app deactivation/reactivation → status transitions correctly + UI banner shows up in all 3 pages

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved handling of Netatmo auth-related HTTP errors (401, 403, 406) to reliably switch to reconnect mode and avoid duplicate reconnect scheduling.
  * Refined service status transitions so “connected” status is only set when not reconnecting/disconnected.
  * Enhanced device discovery/details loading to trigger auth-error handling on failed API calls.
* **New Features**
  * Added optional status-loading support to the Netatmo discover page so it can refresh status after discovery actions.
* **Tests**
  * Expanded unit tests for reconnection idempotency, discovery/status preservation, and auth error propagation across API flows.

### CODE REVIEW

File: server/services/netatmo/lib/netatmo.init.js [hunk 1]
```diff
@@ -1,17 +1,31 @@
...
+    if (response.success) {
+      await this.pollRefreshingToken();
+      await this.refreshNetatmoValues();
+      await this.pollRefreshingValues();
+    }
+  } catch (e) {
+    if (e && e.transient) {
+      logger.warn('Netatmo init: refresh failed transiently, scheduling auto-reconnect: ', e.message);
+      this.scheduleReconnectAttempt();
+      return;
```
[the remark this answers is not in the document]
Comment (Terdious) [reply server/services/netatmo/lib/netatmo.init.js:26]: Addressed in `netatmo.pollRefreshingToken.js` rather than `init.js`: the bootstrap of value polling happens in the reconnect scheduler's success path, which is the single place that's reused by both `init()` and the periodic poll tick. On retry success, after `pollRefreshingToken()`, we now bootstrap values if `this.pollRefreshValues` is not set:

```js
if (!this.pollRefreshValues) {
  await this.refreshNetatmoValues();
  await this.pollRefreshingValues();
}
```
Two dedicated tests cover both cases (should bootstrap value polling on retry success when not already started / should not bootstrap value polling on retry success when already started).

**Comment by Terdious** [comment]
## Update — runtime validation feedback

Following review feedback and runtime testing, the following has been pushed (force-push, 1 commit on the branch):

### Backend fixes
1. **`handleApiAuthError.js`** — `saveStatus(RECONNECTING)` moved **before** the idempotence check. Reason: in the scheduler success path, the second call from `loadDevices` (still hitting 401/403/406 because the Netatmo app remains deactivated) was idempotent → no `saveStatus` → status stayed on `GET_DEVICES_VALUES` → the caller flipped it back to `CONNECTED`. Now the status is always flipped to `RECONNECTING`; only the scheduling stays idempotent.
2. **`pollRefreshingToken.js`** — In the scheduler success path, after `refreshNetatmoValues()`, detect `status === RECONNECTING` and re-schedule a new reconnect. Reason: Netatmo's `/oauth2/token` endpoint accepts the refresh even when the app is deactivated, so the OAuth refresh "succeeds" but business API calls still fail. Without this re-detection, the UI was flickering between `connected` and `reconnecting`.
3. **`pollRefreshingToken.js`** — Bootstrap value polling on retry success when not already started (CodeRabbit feedback #1).

### Frontend fixes
4. **`discover-page/index.js`** — `removeListener(ERROR.CONNECTING, updateStatus)` → `updateStatusError` (latent bug, the `addListener` had used `updateStatusError`). Reduces listener leak across navigations.
5. **`discover-page/index.js` + `DiscoverTab.jsx`** — `loadStatus` passed as a prop and called after `getDiscoveredDevices` / `refreshDiscoveredDevices`. Defense-in-depth: forces an HTTP resync of the status after each user action on the Discover page.

### Test fixes (CodeRabbit feedback)
- 4× `try/finally` around `sinon.spy.restore()` in the new auth-error tests (#2, #3, #4)
- Explicit `clearTimeout` assertions in the new timeout-cleanup tests in `saveStatus.test.js` (#5)

### Known limitation (Discover UI)

When the discovered devices list is already populated on the Discover page, clicking ""Refresh"" while the Netatmo app is deactivated **does not visually update the RECONNECTING banner** on that page. The backend status is correctly switched to `reconnecting` (other Netatmo pages — Devices, Setup — display the banner properly, and a page reload on Discover also shows it).

Runtime DevTools investigation revealed multiple zombie instances of `NetatmoDiscoverPage` receiving the WebSocket event (each event arrives 4×), suggesting a `componentWillUnmount` / `removeListener` issue at the preact-router level — possibly aggravated by HTML-validity warnings on `<p>` elements containing `<div>` (via `<MarkupText>` i18n content) that break Preact's reconciliation on that specific page.

Investigation didn't yield a low-risk fix within the scope of this PR. The maintainer (@Terdious) considers this a low-priority edge case for a follow-up PR if it ever bothers users in practice.

### Validation
- `npm run test-service --service=netatmo` → 201 passing
- 100% coverage on all touched server files
- Runtime validated for the main scenarios: F5 on any Netatmo page after app deactivation → RECONNECTING banner shows; auto-reconnect works when the app is reactivated; Devices and Setup pages update live via WebSocket.

**Review by Pierre-Gilles — APPROVED** [review]
Good for me! Thanks!
